File: services/payment-service/src/main/java/com/example/barbooking/payment/domain/model/Payment.java
package com.example.barbooking.payment.domain.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "payments")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long bookingId;

    @Column(nullable = false)
    private Double amount;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private PaymentStatus status = PaymentStatus.PENDING;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private PaymentMethod paymentMethod;

    @Column(unique = true, length = 100)
    private String transactionId;

    @Column
    private String failureReason;

    @Column
    private LocalDateTime paidAt;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(nullable = false)
    private LocalDateTime updatedAt;

    public enum PaymentStatus {
        PENDING,
        PROCESSING,
        COMPLETED,
        FAILED,
        REFUNDED
    }

    public enum PaymentMethod {
        CREDIT_CARD,
        DEBIT_CARD,
        MOBILE_BANKING,
        QR_CODE,
        CASH
    }

    // Business logic methods
    public void markAsCompleted(String transactionId) {
        if (this.status != PaymentStatus.PENDING && this.status != PaymentStatus.PROCESSING) {
            throw new IllegalStateException("Can only complete pending or processing payments");
        }
        this.status = PaymentStatus.COMPLETED;
        this.transactionId = transactionId;
        this.paidAt = LocalDateTime.now();
    }

    public void markAsFailed(String reason) {
        if (this.status == PaymentStatus.COMPLETED || this.status == PaymentStatus.REFUNDED) {
            throw new IllegalStateException("Cannot fail completed or refunded payments");
        }
        this.status = PaymentStatus.FAILED;
        this.failureReason = reason;
    }

    public void markAsRefunded() {
        if (this.status != PaymentStatus.COMPLETED) {
            throw new IllegalStateException("Can only refund completed payments");
        }
        this.status = PaymentStatus.REFUNDED;
    }
}

File: frontend/types/payment.ts
// Payment related types

export type PaymentStatus = 'PENDING' | 'PROCESSING' | 'COMPLETED' | 'FAILED' | 'REFUNDED';
export type PaymentMethod = 'CREDIT_CARD' | 'DEBIT_CARD' | 'MOBILE_BANKING' | 'QR_CODE';

export interface Payment {
  id: number;
  bookingId: number;
  amount: number;
  status: PaymentStatus;
  paymentMethod: PaymentMethod;
  transactionId: string | null;
  failureReason: string | null;
  paidAt: string | null;
  createdAt: string;
}

export interface InitiatePaymentRequest {
  bookingId: number;
  amount: number;
  paymentMethod: PaymentMethod;
}

export interface PaymentMethodInfo {
  method: PaymentMethod;
  label: string;
  labelTh: string;
  icon: string;
  description: string;
}

export const PAYMENT_METHOD_INFO: Record<PaymentMethod, PaymentMethodInfo> = {
  CREDIT_CARD: {
    method: 'CREDIT_CARD',
    label: 'Credit Card',
    labelTh: 'บัตรเครดิต',
    icon: '💳',
    description: 'ชำระด้วยบัตรเครดิต'
  },
  DEBIT_CARD: {
    method: 'DEBIT_CARD',
    label: 'Debit Card',
    labelTh: 'บัตรเดบิต',
    icon: '💳',
    description: 'ชำระด้วยบัตรเดบิต'
  },
  MOBILE_BANKING: {
    method: 'MOBILE_BANKING',
    label: 'Mobile Banking',
    labelTh: 'โมบายแบงก์กิ้ง',
    icon: '📱',
    description: 'ชำระผ่านแอปธนาคาร'
  },
  QR_CODE: {
    method: 'QR_CODE',
    label: 'QR Code',
    labelTh: 'คิวอาร์โค้ด',
    icon: '📱',
    description: 'สแกน QR Code เพื่อชำระเงิน'
  }
};

